packages/compass-e2e-tests/tests/data-modeling-tab.test.ts

```diff
@@ -14,6 +14,28 @@ import {
   createNumbersCollection,
 } from '../helpers/insert-data';
 
+type DiagramInstance = {
+  getNodes: () => Array<{ id: string; data: { title: string } }>;
+};
+
+async function getDiagramNodes(browser: CompassBrowser) {
+  return await browser.execute(() => {
+    // eslint-disable-next-line no-restricted-globals
+    if ('diagramInstance' in window) {
+      // eslint-disable-next-line no-restricted-globals
+      const diagramInstance = window.diagramInstance as DiagramInstance;
+      const nodes = diagramInstance.getNodes();
+      return nodes.map((node) => ({
+        id: node.id,
+        title: node.data.title,
+      }));
+    }
+    throw new Error(
+      'Diagram instance not found in the window object. Ensure the diagram is set for tests.'
+    );
+  });
+}
```

Comment:
Can't say I have the strongest preference, but these are rendering real DOM nodes on the screen and we do expect them to be visible (and so accessible through the normal wdio APIs). Are there any issues with testing them without relying on internals of the library? These are e2e tests after all

Reply:
Currently we are not positioning these on the viewport correctly (the nodes are randomly placed) and with virutalization enabled for diagramming, they many not be rendered. So I tried to use this approach. 

I think for tests we may want to disable virtualization so that everything is rendered and we don't need to do this kinda stuff or even using ref to access diagram. I added [COMPASS-9447](https://jira.mongodb.org/browse/COMPASS-9447) to export virtualization flag from diagramming package and once that's done, I can do a followup on this and clean it completely. 

Let me know what you think about it?

Reply:
(maybe with failing tests on mac, i need to revisit this whole thing in this PR actually)